File: backend/src/main/java/com/hehe/thesocial/dto/response/feed/FeedItemResponse.java
package com.hehe.thesocial.dto.response.feed;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.hehe.thesocial.dto.response.file.FileResponse;
import com.hehe.thesocial.dto.response.userDetail.UserDetailResponse;
import com.hehe.thesocial.entity.enums.FeedItemType;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class FeedItemResponse {
    String id;
    FeedItemType feedItemType;
    FileResponse video;
    ImageSlideResponse imageSlide;
    String title;
    String description;
    Set<String> hashTagIds;
    Set<String> commentIds;
    long likeCount;
    long commentCount;
    long shareCount;
    long viewCount;
    boolean active;
    boolean violated;
    LocalDateTime disabledAt;
    UserDetailResponse disabledBy;
    LocalDateTime createdAt;
    LocalDateTime updatedAt;
    boolean loved;
    Set<String> hashTags;
    UserDetailResponse uploader; // Thông tin người upload
}

File: mobile-app/src/Types/response/FeedItemResponse.ts
import FileResponse from "./FileResponse";
import { UserDetailResponse } from "./UserDetailResponse";

export enum FeedItemType {
  VIDEO = "VIDEO",
  IMAGE_SLIDE = "IMAGE_SLIDE",
}

export interface ImageSlideResponse {
  id: string;
  images: FileResponse[];
  createdAt: string;
  updatedAt: string;
}

export default interface FeedItemResponse {
  id: string;
  feedItemType: FeedItemType;
  video: FileResponse | null;
  imageSlide: ImageSlideResponse | null;
  title?: string;
  description?: string;
  hashTagIds?: string[];
  hashTags?: string[];
  commentIds?: string[];
  likeCount: number;
  commentCount: number;
  shareCount: number;
  createdAt: string;
  updatedAt: string;
  loved?: boolean;
  uploader?: UserDetailResponse; // Thông tin người upload
}
